etcdserver/server.go

```diff
@@ -612,8 +615,18 @@ func (s *EtcdServer) run() {
 			sched.Schedule(f)
 		case leases := <-expiredLeaseC:
 			go func() {
-				for _, l := range leases {
-					s.LeaseRevoke(context.TODO(), &pb.LeaseRevokeRequest{ID: int64(l.ID)})
+				// Increases throughput of exipired leases deletion process through parallization
+				c := make(chan struct{}, goroutinesCountForProcessingExpiredLeases)
+				for _, lease := range leases {
+					select {
+					case c <- struct{}{}:
+					case <-s.done:
+						return
+					}
+					go func() {
+						s.LeaseRevoke(context.TODO(), &pb.LeaseRevokeRequest{ID: int64(lease.ID)})
+						<-c
```

Comment:
`<-c` would make this goroutine block

Can we do the other way? Send events to buffered channel when `LeaseRevoke` is done?